src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsSpreadsheet.kt

```diff
@@ -4,19 +4,26 @@ import org.apache.poi.ss.usermodel.Cell
 import org.apache.poi.ss.usermodel.Row
 import org.apache.poi.ss.usermodel.Workbook
 import uk.gov.justice.digital.hmpps.pecs.jpc.location.Location
+import uk.gov.justice.digital.hmpps.pecs.jpc.location.LocationRepository
 import uk.gov.justice.digital.hmpps.pecs.jpc.location.LocationType
 import java.io.Closeable
 
 private const val TYPE = 1
 private const val SITE = 2
 private const val AGENCY = 3
 private const val COLUMN_HEADINGS = 1
-private const val ROW_OFFSET = 2
+private const val ROW_OFFSET = 1
 
 /**
  * Simple wrapper class to encapsulate the logic around access to data in the locations spreadsheet. When finished with the spreadsheet should be closed.
  */
-class LocationsSpreadsheet(private val spreadsheet: Workbook) : Closeable {
+class LocationsSpreadsheet(private val spreadsheet: Workbook, private val locationRepository: LocationRepository) : Closeable {
+
+    init {
+        Tab.values().toList().filter { spreadsheet.getSheet(it.label) == null }.joinToString { it.label }.let {
+            if (it.isNotBlank()) throw IllegalStateException("The following tabs are missing from the locations spreadsheet: $it")
```

Comment:
I think you can do `.takeIf{it.IsNotBlank()}.let{throw Illegal ....}`

or some such syntax

Reply:
Will `takeIf { it.works } `....I'll get my coat.